classes/local/skel/base.php

```diff
@@ -64,6 +64,15 @@ public function set_data(array $data) {
         $this->data = $data;
     }
 
+    public function set_attribute($attribute) {
```

Comment:
I wanted to mention here that I really like how this API reads in the code like
```
$this->files['lib.php']->set_attribute('has_navigation');
```
It is a very elegant and intuitive way to set file-level flags that can be used in templates. Good concept, good naming. Well done.